core/src/main/java/org/apache/iceberg/TableProperties.java

```diff
@@ -228,4 +228,10 @@ private TableProperties() {
 
   public static final String MERGE_CARDINALITY_CHECK_ENABLED = "write.merge.cardinality-check.enabled";
   public static final boolean MERGE_CARDINALITY_CHECK_ENABLED_DEFAULT = true;
+
+  public static final String WATERMARK_FIELD_NAME = "write.watermark.field";
+  public static final String WATERMARK_FIELD_NAME_DEFAULT = "";
+
+  public static final String WATERMARK_VALUE = "write.watermark";
```

Comment:
In the past, we have 3 Flink streaming jobs (1 for each AWS region) writing to the same table. We need to write to 3 different watermark table properties (1 for each region). Watermark consumer then use the min value to determine the overall table watermark. 

A provider pattern similar to `WRITE_LOCATION_PROVIDER_IMPL` can work. A default impl could be a single property name. not sure if there is a simpler way to achieve this.